sabnzbd/filesystem.py

```diff
@@ -58,6 +58,23 @@ def get_ext(filename: str) -> str:
         return ""
 
 
+def has_unwanted_extension(filename: str) -> bool:
+    """ Determine if a filename has an unwanted extension, given the configured mode """
+    extension = get_ext(filename).replace(".", "").lower()
+    if extension and sabnzbd.cfg.unwanted_extensions():
+        return (
+            # Blacklisted
+            sabnzbd.cfg.unwanted_extensions_mode() == 0
+            and extension in sabnzbd.cfg.unwanted_extensions()
+        ) or (
+            # Not whitelisted
+            sabnzbd.cfg.unwanted_extensions_mode() == 1
+            and extension not in sabnzbd.cfg.unwanted_extensions()
+        )
+    else:
+        return bool(sabnzbd.cfg.unwanted_extensions_mode())
```

Comment:
I am not sure I get this `else`? What case is it triggered?

Reply:
No extension in the filename and/or no extensions configured in the unwanted extension setting. In those cases the mode is basically all we have to go with.